### ISSUE TITLE
[BUG REPORT] Throwing simc overflow error to user

### ISSUE THREAD

**Issue by Math-O5** [cimplec/sim-c#361]
**Describe the bug**
When user type incomplete programs, the "check_if[i + 1]" will always overflow. 

The sintax is wrong, but the program should treat this properly. 

**To Reproduce**
Complie this simc code:

test.simc:

```
MAIN
    if()
```

```
MAIN
    var
```

```
MAIN
    f
```

```
MAIN
    array overflow
```

**Expected behavior**
A kind message error for user

**Desktop (please complete the following information):**
 - OS: Fedora 30, Ubuntu 18

**Comment by frankhart2018** [comment]
Yeah currently it just expects something to be there, there should be an index out of range check too in check_if.

**Comment by frankhart2018** [comment]
Passing the index as argument and processing check_if only if the index is in range will fix this.

**Comment by jigyasudhingra** [comment]
I would like to work on this @frankhart2018.

**Comment by Math-O5** [comment]
@jigyasudhingra All yours. :)

**Comment by lolzone13** [comment]
May I be assigned, I think I have fixed this.

**Comment by ami-harry** [comment]
your list is empty and you are trying to compare its index type, but your list is empty so that it is giving the output like that.

**Comment by frankhart2018** [comment]
Do you want to work in this @ami-harry

**Comment by ami-harry** [comment]
@frankhart2018  yes. I want to work

**Comment by frankhart2018** [comment]
Sure assigning this to you too @ami-harry.
